spec/helpers/faker/puppet_module_remote_repo.rb

```diff
@@ -0,0 +1,130 @@
+require 'open3'
+
+require_relative '../faker'
+
+# Fake a remote git repository that hold a puppet module
+#
+# This module allow to fake a remote repositiory using:
+#  - a bare repo
+#  - a temporary cloned repo to operate on the remote before exposition to modulesync
+#
+# Note: This module need to have working_directory sets before using it
+module Faker
+  class PuppetModuleRemoteRepo
+    attr_reader :name
+    attr_reader :namespace
+
+    def initialize(name, namespace)
+      @name = name
+      @namespace = namespace
+    end
+
+    def populate
+      FileUtils.chdir(Faker.working_directory) do
+        run "git init --bare '#{bare_repo_dir}'"
+        run "git clone '#{bare_repo_dir}' '#{tmp_repo_dir}'"
+
+        module_short_name = name.sub(/^puppet-/, '')
+
+        FileUtils.chdir(tmp_repo_dir) do
+          metadata = {
+            name: "modulesync-#{module_short_name}",
+            version: '0.4.2',
+            author: 'ModuleSync team',
+          }.to_json
+
+          File.write 'metadata.json', metadata
```

Comment:
Just nitpicking here. What do you think about moving the `.to_json` to the `File.write` line? That way `metadata` is a pure datastructure and serializing is clearly done only for writing to the file.

Reply:
Done.